packages/odyssey-react/src/components/TextInput/TextInput.module.scss

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+.textInput {
+  @include input-baseline;
+
+  &[type='search'] {
+    padding-inline-start: ms(1) + $spacing-s + $spacing-xs;
+    background-image: get-icon('search', $text-body);
+    background-repeat: no-repeat;
+    background-position: left $spacing-s center;
+    background-size: ms(1);
+
+    &::-webkit-search-decoration,
+    &::-webkit-search-cancel-button,
+    &::-webkit-search-results-button,
+    &::-webkit-search-results-decoration {
+      display: none;
+    }
+
+    &::-ms-clear,
+    &::-ms-reveal {
+      display: none;
+    }
+
+    [dir='rtl'] & {
+      background-position: right $spacing-s center;
+    }
+  }
+}
+
+.optionalLabel {
+  display: block;
+  color: $text-sub;
+  font-size: $size-body-caption;
+  font-weight: 400;
+  line-height: $spacing-m;
+}
+
+.fieldset {
+  margin-block-start: 0;
+  margin-block-end: $spacing-s;
+  margin-inline: 0;
+
+  &:last-child {
+    margin-block: 0;
+    margin-inline: 0;
+  }
+}
+
+.label {
+  display: flex;
+  align-items: flex-end;
+  justify-content: space-between;
+  color: $text-heading;
+  font-weight: 600;
+}
+
+.labelDisabled {
+  composes: label;
```

Comment:
Great work! Would it be helpful if I looked into linting rules available for ordering or separating out `composes`? Might be helpful since it looks like a normal CSS property.

Reply:
Good thought I can look into making sure it gets an extra space after it